crates/ika-core/src/dwallet_mpc/mpc_outputs_verifier.rs

```diff
@@ -202,9 +166,41 @@ impl DWalletMPCOutputsVerifier {
         })
     }
 
-    fn epoch_store(&self) -> DwalletMPCResult<Arc<AuthorityPerEpochStore>> {
-        self.epoch_store
-            .upgrade()
-            .ok_or(DwalletMPCError::EpochEnded(self.epoch_id))
+    /// Syncs the [`DWalletMPCOutputsVerifier`] from the epoch start.
+    /// Fails only if the epoch switched in the middle of the state sync (in which case the process is exited and this struct would be re-initialized).
+    pub fn bootstrap_from_storage(&mut self, epoch_store: &AuthorityPerEpochStore) -> IkaResult {
+        info!("Bootstrapping MPC Outputs Verifier from Storage");
+        for output in epoch_store.tables()?.get_all_dwallet_mpc_outputs()? {
+            let party_to_authority_map = epoch_store.committee().party_to_authority_map();
+            let mpc_protocol_name = output.session_info.mpc_round.to_string();
+
+            // Create a base logger with common parameters.
+            let base_logger = MPCSessionLogger::new()
+                .with_protocol_name(mpc_protocol_name.clone())
+                .with_party_to_authority_map(party_to_authority_map.clone());
+            let session_identifier = CommitmentSizedNumber::from_le_slice(
+                &output.session_info.session_identifier.into_bytes(),
+            );
+            base_logger.write_output_to_disk(
+                session_identifier,
+                epoch_store.authority_name_to_party_id(&epoch_store.name)?,
+                epoch_store.authority_name_to_party_id(&output.authority)?,
+                &output.output,
+                &output.session_info,
+            );
+            if let Err(err) = self.try_verify_output(
```

Comment:
inside this, it will write the session id into a c hannel, and we will read it in the mpc service before we spawn, and create a `FINISHED` session for it